app/javascript/react/screens/App/Settings/screens/ConversionHostsSettings/components/ConversionHostWizard/ConversionHostWizardLocationStep/__tests__/ConversionHostWizardLocationStep.test.js

```diff
@@ -9,6 +10,10 @@ import { RHV, OPENSTACK } from '../../../../../../../../../../common/constants';
 describe('conversion host wizard location step', () => {
   const store = generateStore({ form: formReducer }, {});
 
+  const ProviderWrapper = (
+    { children, ...props } // eslint-disable-line react/prop-types
+  ) => <Provider store={store}>{cloneElement(children, props)}</Provider>;
```

Comment:
Well, `setProps` might not be the best way to do it, but in those examples I was trying to just test the behavior of `componentDidUpdate` as an isolated unit, and changing the props is an easy way to trigger that method and assert its behavior in response to those props.

What I was trying to do in a lot of these tests is test the inner component itself as a unit without the context of Redux, and not the connected component, just to make sure it behaves the way it should when given the props it is given. That's why I import their file directly and not the `index.js` that wraps them in `connect()` (that gets its own unit test, which just makes sure the right props are selected from redux and passed to the component). The problem here is that the components using redux-form are all exported with the `reduxForm()` call right inline at the bottom of the file, so there isn't a way to import them into the tests without that. I think the real fix for unit testing these would be to move `reduxForm()` into the `index.js` file, it's part of connecting the components to redux anyway, I think that's just a mistake we made early on that is getting duplicated. And in fact, the components are getting connected to redux twice `<Connect(Connect(Form(...)))>` because of the `connect()` in index.js wrapping the `reduxForm()` in the component file. That's probably also something we could clean up with hooks.

> I would like to add that is not exactly the best practice and props should be changed via the actual event which is supposed to mutate them in UI (some element click or store state change etc.)

It sounds like what you're advocating for is an integration test for the whole connected component, which I agree would be a good thing to add, we don't do that enough in this repo. I do think the more granular unit tests are still useful, though.

This is giving me a lot to think about for sure! 😄 I think when we eventually rewrite all these complex components with hooks, we can certainly write better tests too and think these things through better.